chain/messagepool/pruning.go

```diff
@@ -14,12 +14,13 @@ import (
 )
 
 func (mp *MessagePool) pruneExcessMessages() error {
-	mp.curTsLk.Lock()
-	ts := mp.curTs
-	mp.curTsLk.Unlock()
+	mp.transactionLk.Lock()
```

Comment:
Also not clear to me that we need `transactionLk` here, but unsure

Reply:
same as above - we are changing the message pool so we want to use the transaction and state locks because we are changing either the tipset or the messages